tests/suites/test_suite_pk.function

```diff
@@ -1924,3 +1971,202 @@ exit:
+    PSA_ASSERT(pk_psa_genkey_generic(PSA_KEY_TYPE_DH_KEY_PAIR(PSA_DH_FAMILY_RFC7919),
+                                     PSA_KEY_USAGE_EXPORT, PSA_ALG_NONE, 2048, &key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(key_id, &pk_ctx), MBEDTLS_ERR_PK_BAD_INPUT_DATA);
+    psa_destroy_key(key_id);
+#endif /* PSA_WANT_KEY_TYPE_DH_KEY_PAIR_GENERATE */
+
+#if defined(MBEDTLS_PK_HAVE_ECC_KEYS) && defined(PSA_WANT_ECC_SECP_R1_256) && \
+    defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE)
+    /* Generate an EC key which cannot be exported. */
+    PSA_ASSERT(pk_psa_genkey_generic(PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1),
+                                     0, PSA_ALG_NONE, 256, &key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(key_id, &pk_ctx), MBEDTLS_ERR_PK_BAD_INPUT_DATA);
+    psa_destroy_key(key_id);
+
+    /* Use valid exportable EC key with wrong alorithm. */
+    PSA_ASSERT(pk_psa_genkey_generic(PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1),
+                                     PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_SIGN_HASH,
+                                     PSA_ALG_ECDH, 256, &key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(key_id, &pk_ctx), MBEDTLS_ERR_PK_BAD_INPUT_DATA);
+    psa_destroy_key(key_id);
+#endif /* MBEDTLS_PK_HAVE_ECC_KEYS && PSA_WANT_ECC_SECP_R1_256 &&
+          PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE */
+
+#if defined(MBEDTLS_RSA_C) && defined(PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE)
+    /* Use valid exportable RSA key with wrong alorithm. */
+    PSA_ASSERT(pk_psa_genkey_generic(PSA_KEY_TYPE_RSA_KEY_PAIR,
+                                     PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_SIGN_HASH,
+                                     PSA_ALG_CMAC, 2048, &key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(key_id, &pk_ctx), MBEDTLS_ERR_PK_BAD_INPUT_DATA);
+    psa_destroy_key(key_id);
+
+    /* Try to encrypt with a RSA key in PKCS1V21 format. */
+#if defined(MBEDTLS_USE_PSA_CRYPTO) && defined(MBEDTLS_MD_CAN_SHA256) && defined(MBEDTLS_PKCS1_V21)
+    PSA_ASSERT(pk_psa_genkey_generic(PSA_KEY_TYPE_RSA_KEY_PAIR,
+                                     PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_ENCRYPT,
+                                     PSA_ALG_RSA_OAEP(PSA_ALG_SHA_256), 2048, &key_id));
+    TEST_EQUAL(mbedtls_pk_copy_from_psa(key_id, &pk_ctx), 0);
+    TEST_EQUAL(mbedtls_pk_encrypt(&pk_ctx, in_buf, sizeof(in_buf),
+                                  out_buf, &out_buf_len, sizeof(out_buf),
+                                  mbedtls_test_rnd_std_rand, NULL),
+               MBEDTLS_ERR_RSA_INVALID_PADDING);
+    psa_destroy_key(key_id);
+#endif /* MBEDTLS_USE_PSA_CRYPTO && MBEDTLS_MD_CAN_SHA256 && MBEDTLS_PKCS1_V21*/
+#endif /* MBEDTLS_RSA_C && PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE */
+
+exit:
+    mbedtls_pk_free(&pk_ctx);
+    psa_destroy_key(key_id);
+    PSA_DONE();
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_PSA_CRYPTO_C*/
+void pk_copy_from_psa_success(data_t *priv_key_data, int key_type_arg,
```

Comment:
What about copying a public key?

Reply:
Isn't this already done when I create `pk_ctx2`?
For that one I start from the private key held on PSA side, I export its public counterpart and I create a new slot still in PSA, then I call `mbedtls_pk_copy_from_psa` on this public key.

Reply:
Ah, right, sorry, I missed that. I've only looked superficially so far, mostly to compare the testing with `import_into_psa`, which I only lately realized is similar.

Please rename `pk_ctx2` to something like `pk_pub`, to make this clearer.